addons/account/migrations/9.0.1.1/post-migration.py

```diff
@@ -303,23 +402,196 @@ def account_partial_reconcile(env):
+    # the company currency. This will dramatically improve the overall
+    # performance of the migration of reconciliations.
+    openupgrade.logged_query(cr, """
+        WITH Q1 AS (
+            SELECT reconcile_id, sum(debit-credit) as balance,
+            count(id) as num_moves, count(currency_id) as num_currencies
+            FROM account_move_line
+            WHERE reconcile_id IS NOT NULL
+            GROUP BY reconcile_id, currency_id
+        ),
+        Q2 AS (
+            SELECT reconcile_id
+            FROM Q1
+            WHERE balance = 0.0
+            AND num_moves = 2
+            AND num_currencies = 0
+        ),
+        Q3 AS (
+            SELECT aml.reconcile_id, aml.id, aml.debit,
+            aml.credit, aml.company_id
+            FROM account_move_line AS aml
+            INNER JOIN Q2
+            ON Q2.reconcile_id = aml.reconcile_id
+        ),
+        Q4 AS (
+            SELECT reconcile_id,
+            CASE WHEN debit > 0.0
+            THEN id ELSE Null END as debit_move_id, CASE WHEN credit > 0.0
+            THEN id ELSE Null END as credit_move_id,
+            CASE WHEN debit > 0.0 THEN debit END as amount, company_id
+            FROM Q3
+            GROUP BY reconcile_id, debit_move_id, credit_move_id, amount,
+            company_id
+        ),
+        Q5 AS (
+            SELECT reconcile_id, sum(debit_move_id) as debit_move_id,
+            sum(credit_move_id) as credit_move_id, sum(amount) as amount,
+            company_id
+            FROM Q4
+            GROUP BY reconcile_id, company_id
+        )
+        INSERT INTO account_partial_reconcile
+        (create_uid, create_date, write_uid, write_date, debit_move_id,
+        credit_move_id, amount, company_id, amount_currency)
+        SELECT amr.create_uid, amr.create_date, amr.write_uid,
+        amr.write_date, Q5.debit_move_id, Q5.credit_move_id, Q5.amount,
+        Q5.company_id, 0.0
+        FROM Q5
+        INNER JOIN account_move_reconcile AS amr
+        ON Q5.reconcile_id = amr.id
+        WHERE debit_move_id > 0 AND credit_move_id > 0
+    """)
+
+    # We want to exclude the moves that were included in the step above from
+    # the next reconciliation step.
+    cr.execute("""
+        WITH Q1 AS (
+            SELECT reconcile_id, sum(debit-credit) as balance,
+            count(id) as num_moves, count(currency_id) as num_currencies
+            FROM account_move_line
+            WHERE reconcile_id IS NOT NULL
+            GROUP BY reconcile_id, currency_id
+        ),
+        Q2 AS (
+            SELECT reconcile_id
+            FROM Q1
+            WHERE balance = 0.0
+            AND num_moves = 2
+            AND num_currencies = 0
+        )
+        SELECT aml.id
+        FROM account_move_line AS aml
+        INNER JOIN Q2
+        ON Q2.reconcile_id = aml.reconcile_id
+    """)
+
+    move_line_ids_reconciled = [move_line_id for move_line_id,
+                                in cr.fetchall()]
```

Comment:
the way I read the code, you never really need this list on the client side. Consider selecting those ids into a temporary table (`create temporary table`), then you don't need to pass those back and forth between your code and the sql server, and added benefit is that you can use this for joins etc if convenient. This might also reduce the memory footprint on the client side